### PR TITLE
feat(sources): deprecate mime_type on add_file path; keep Drive --mime-type (T6.E)

### PR DESCRIPTION
**Pull request by teng-lin**
## Summary

Tier-6 PR-T6.E from `.sisyphus/plans/tier-6-implementation.md`. The `mime_type` argument on `SourcesAPI.add_file` and the `--mime-type` flag on the file-source CLI path were never wired into the resumable-upload RPC payload (the server derives the MIME type from the filename extension). Both now emit a deprecation signal; the separate Drive `--mime-type` on `source add-drive` is **live** and intentionally untouched.

## Changes

- **`src/notebooklm/_sources.py`** — `add_file(..., mime_type=...)` emits `DeprecationWarning` (stacklevel=2) when a non-None value is passed. Positional signature preserved for backward compatibility. `# DEPRECATION-REMOVAL: v0.X.0` marker placed.
- **`src/notebooklm/cli/source.py`** — file-source path emits `click.echo(..., err=True)` deprecation note when `--mime-type` is used with `--type=file` (or auto-detected file source). Suppressible via `NOTEBOOKLM_QUIET_DEPRECATIONS=1`. The CLI no longer forwards the unused arg to `add_file`, so the user gets exactly one signal.
- **`src/notebooklm/cli/source.py:697-715`** — Drive `--mime-type` flow **NOT TOUCHED**. It remains the canonical way to pick `google-doc` / `google-slides` / `google-sheets` / `pdf`.
- **`CHANGELOG.md`** — new `### Deprecated` section under `[Unreleased]` covering both the library and CLI deprecations, with migration guidance.

## Tests

- `tests/unit/test_sources_upload.py`:
  - `test_add_file_mime_type_non_none_emits_deprecation_warning` — `pytest.warns(DeprecationWarning, match="mime_type")`.
  - `test_add_file_mime_type_none_does_not_warn` — default path stays silent.
- `tests/unit/cli/test_source.py`:
  - `test_source_add_file_with_mime_type_emits_deprecation` — stderr contains "unused for file sources"; CLI does NOT forward `mime_type` to `add_file`.
  - `test_source_add_file_mime_type_suppressed_by_env` — `NOTEBOOKLM_QUIET_DEPRECATIONS=1` suppresses.
  - `test_source_add_drive_mime_type_no_deprecation_warning` — regression guard across all four Drive choices.
- `tests/integration/test_sources.py::test_add_epub_file_upload` — dropped the now-deprecated `mime_type="application/epub+zip"` kwarg so the integration suite stops leaking a `DeprecationWarning`.

## Test plan

- [x] `uv run ruff format .`
- [x] `uv run ruff check .`
- [x] `uv run mypy src/notebooklm --ignore-missing-imports`
- [x] `uv run pytest tests/unit -x -q` — 2707 passed, 7 skipped
- [x] `uv run pytest tests/ -x -q --ignore=tests/e2e` — 3306 passed, 11 skipped
- [ ] CI green on the PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Deprecations**
  * MIME type parameter for file uploads deprecated; MIME now inferred from filename extensions.
  * CLI `--mime-type` flag deprecated for file-source uploads; remains functional for Drive uploads.

* **Documentation**
  * CLI reference updated to document the deprecation and suppression option.
  * Added environment variable to suppress deprecation notices.

* **Tests**
  * Added/updated tests to assert deprecation notices and suppression behavior for CLI and upload flows.

[Review Change Stack](https://app.coderabbit.ai/change-stack/teng-lin/notebooklm-py/pull/487)

### CODE REVIEW

**Comment by teng-lin** [comment]
@claude review

**Comment by teng-lin** [comment]
Thanks for the review! Addressed in `a53f7e0`:

- **`NOTEBOOKLM_QUIET_DEPRECATIONS` documentation gap** — added a dedicated subsection to `docs/configuration.md` (`### NOTEBOOKLM_QUIET_DEPRECATIONS`) explaining the truthy-non-empty semantics and the note that library-level `DeprecationWarning`s are unaffected. Also added a `### Source: add — --mime-type deprecation` callout to `docs/cli-reference.md` showing the deprecated vs migrated invocation and linking back to the env var docs.
- **`import warnings as _warnings` inside the test body** — hoisted to module-level imports in `tests/unit/test_sources_upload.py` and renamed back to plain `warnings`.
- **CLI `result.output` relies on implicit `mix_stderr=True`** — added an explicit comment on `test_source_add_file_with_mime_type_emits_deprecation` calling out the dependency so a future fixture change to `mix_stderr=False` won't silently turn the test into a no-op.

Skipped:
- **Silent no-op for non-file types with `--mime-type`** — agreed it's an edge case; the user clearly wanted the file flag (which is now deprecated), so warning the user when they pass `--mime-type` with `--type text` would over-fire. Tracking this only via the file-source gate keeps the signal targeted.
- **`v0.X` version placeholder** — left as-is for now; will be filled in by the removal commit. The `# DEPRECATION-REMOVAL: v0.X.0` marker is the tracker as you suggested.

File: src/notebooklm/cli/source.py [hunk 1]
```diff
@@ -384,6 +385,11 @@ async def _run():
     help="Source type (auto-detected if not specified)",
 )
 @click.option("--title", help="Custom title for text and uploaded-file sources")
+# DEPRECATION-REMOVAL: v0.X.0 — ``--mime-type`` on the file-source path is a
+# no-op (the upload pipeline ignores it; the server derives the MIME type from
+# the filename extension). A deprecation note is echoed to stderr when the flag
+# is used with a file source. The separate Drive-source ``--mime-type`` on the
+# ``add-drive`` command remains live and IS NOT affected by this deprecation.
 @click.option("--mime-type", help="MIME type for file sources")
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/cli/source.py]: Addressed in 05457e4: CLI help text now leads with `[Deprecated]` so the flag status surfaces in `--help`. The library-side `warnings.warn` already exists in `_sources.py:add_file`; the CLI uses a `click.echo(..., err=True)` notice instead because Click suppresses Python warnings emitted from CLI handlers.

File: src/notebooklm/cli/source.py [hunk 2]
```diff
@@ -467,6 +473,21 @@ def source_add(
...
         upload_path = _validate_upload_path(content, follow_symlinks)
 
+    # DEPRECATION-REMOVAL: v0.X.0 — ``--mime-type`` is a no-op on the file
+    # source path (the upload pipeline ignores it). The Drive ``--mime-type``
+    # on ``source add-drive`` is untouched. Suppressible via
+    # ``NOTEBOOKLM_QUIET_DEPRECATIONS=1`` so CI logs stay clean.
+    if (
+        mime_type is not None
+        and detected_type == "file"
+        and not os.environ.get("NOTEBOOKLM_QUIET_DEPRECATIONS")
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/cli/source.py]: Addressed in 05457e4: changed to exact `== "1"` so `NOTEBOOKLM_QUIET_DEPRECATIONS=0` (or `false`) no longer silently suppresses the notice. Updated docs/configuration.md to match.

File: src/notebooklm/_sources.py [hunk 3]
```diff
@@ -545,6 +550,19 @@ async def add_file(
...
         logger.debug("Adding file source to notebook %s: %s", notebook_id, file_path)
+        # DEPRECATION-REMOVAL: v0.X.0 — the ``mime_type`` argument is unused by
+        # the resumable-upload pipeline (the server derives the MIME type from
+        # the filename extension). Kept as a positional kwarg for backward
+        # compatibility; callers passing a non-None value get a warning. See
+        # CHANGELOG entry "Deprecated: ``SourcesAPI.add_file`` ``mime_type``
+        # parameter" under ``[Unreleased]`` for the planned removal release.
+        if mime_type is not None:
+            warnings.warn(
+                "mime_type parameter is unused and will be removed in v0.X; "
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/_sources.py]: Addressed in 05457e4: warning text now reads `v0.X.0` to match the `# DEPRECATION-REMOVAL: v0.X.0` markers + docstring throughout the file.

**Comment by teng-lin** [comment]
@claude review

File: docs/cli-reference.md [hunk 4]
```diff
@@ -553,6 +553,32 @@ notebooklm --profile work auth refresh --quiet
...
+notebooklm source add ./report.pdf --type file --mime-type application/pdf
+
+# Migrated — drop the flag
+notebooklm source add ./report.pdf --type file
+```
+
+To suppress the stderr notice (useful for CI logs where the message would
+repeat across pipeline invocations), set `NOTEBOOKLM_QUIET_DEPRECATIONS=1`
+(any non-empty value). See
+[configuration.md#notebooklm_quiet_deprecations](configuration.md#notebooklm_quiet_deprecations).
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply docs/cli-reference.md:575]: Addressed in 7496d02: docs/cli-reference.md now says `exactly \`1\`` (matching docs/configuration.md and the code which uses `== "1"`). All three sites agree.

**Comment by teng-lin** [comment]
@claude review
